### PR TITLE
Phases out column of train table in info tab now hiden when the game doesn't phase out trains (fixes# 1444)

### PR DESCRIPTION
**Pull request by captianobi**
image
Now
image
Before.

Sorry about the fact that it claims that it's two files changed it's only really one its just github is being weird.

### CODE REVIEW

File: assets/app/view/game/game_info.rb [hunk 1]
```diff
@@ -149,16 +149,21 @@ def upcoming_trains
...
-            h(:td, names_to_prices.keys.join(', ')),
-            h('td.right', names_to_prices.values.map { |p| @game.format_currency(p) }.join(', ')),
-            h(:td, trains.size),
-            h(:td, obsolete_schedule[name]&.join(', ') || 'None'),
-            h(:td, rust_schedule[name]&.join(', ') || 'None'),
-            h(:td, discounts&.join(' ')),
-            h(:td, train.available_on),
-            h(:td, event_text.join(', ')),
-          ])
+          upcoming_train_content = []
```
Comment (tobymao) [inline assets/app/view/game/game_info.rb:152]: change it to 

upcoming_trains_content = [
a, 
b, 
c,
]

upcoming_train << x if y

ucoming_train.concat([
d,
e,
f]

File: assets/app/view/game/game_info.rb [hunk 2]
```diff
@@ -177,21 +182,25 @@ def upcoming_trains
           ])]
         end
 
+        upcoming_train_header = []
+        upcoming_train_header << h(:th, 'Type')
+        upcoming_train_header << h(:th, 'Price')
+        upcoming_train_header << h(:th, 'Remaining')
+        upcoming_train_header << h(:th, 'Phases out') if obsolete_schedule.any? { |key, _value| !key.nil? }
```
[the remark this answers is not in the document]
Comment (captianobi) [reply assets/app/view/game/game_info.rb:189]: For point number 2 it can't ruby counts nil as a key

[the remark this answers is not in the document]
Comment (tobymao) [reply assets/app/view/game/game_info.rb:189]: i don't understand what the issue is

[the remark this answers is not in the document]
Comment (captianobi) [reply assets/app/view/game/game_info.rb:189]: Ruby will count nil: D etc as a valid key and by default everything is assigned to nil

**Comment by captianobi** [comment]
I think I've fixed all the issues with this code.

File: assets/app/view/game/game_info.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (tobymao) [reply assets/app/view/game/game_info.rb]: [nil].any? is false

File: assets/app/view/game/game_info.rb [hunk 3]
```diff
@@ -127,6 +127,8 @@ def upcoming_trains
           obsolete_schedule[first.obsolete_on] = Array(obsolete_schedule[first.obsolete_on]).append(name)
         end
 
+        show_obsolete_schedule = obsolete_schedule.any? { |key, _value| !key.nil? }
```
Comment (tobymao) [inline assets/app/view/game/game_info.rb:130]: this should be

show_obsolete_schedule = obsolete_schedule.keys.any?

File: assets/app/view/game/game_info.rb (hunk 1 above)
Comment (tobymao) [reply assets/app/view/game/game_info.rb:154]: can you make it 

upcoming_train_content = [
...
]

File: assets/app/view/game/game_info.rb [hunk 4]
```diff
@@ -177,21 +183,26 @@ def upcoming_trains
           ])]
         end
 
+        upcoming_train_header = []
```
Comment (tobymao) [inline assets/app/view/game/game_info.rb:186]: fix this too

File: assets/app/view/game/game_info.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (captianobi) [reply assets/app/view/game/game_info.rb]: Nope, because permenets are marked as obsoleted by nil

[the remark this answers is not in the document]
Comment (captianobi) [reply assets/app/view/game/game_info.rb]: Nope, because permenets are marked as obsoleted by nil

**Review by tobymao — APPROVED** [review]

**Pull request merged**